tests/js/client/shell/transaction/shell-transaction.inc

```diff
@@ -5719,6 +5719,20 @@ function transactionDatabaseSuite() {
         trx1.abort();
       }
     },
+
+    testTransactionsAreDatabaseLocal: function() {
+      db._useDatabase(dbn1);
+      const opts = { collections: { write: [cn] } };
+      const trx = db._createTransaction(opts);
+      trx.collection(cn).insert({ _key: "test" });
+      try {
+        db._useDatabase(dbn2);
+        trx.collection(cn).document("test");
+        fail();
+      } catch(err) {
+        assertEqual(internal.errors.ERROR_TRANSACTION_NOT_FOUND.code, err.errorNum);
+      }
```

Comment:
Should there be a `finally` clause here to switch back to the original database?
Otherwise the test case may reset some global state (i.e. the current database) and thus have side-effects for following tests. (note: there are currently no tests failing the new one, but I consider it good practice if a test cleans up after itself).

Reply:
Good point! 👍